apps/blockchain/lib/blockchain/account/repo/cache.ex

```diff
@@ -129,7 +129,10 @@ defmodule Blockchain.Account.Repo.Cache do
     else
       state_with_account = Account.put_account(state, address, account)
 
-      if code, do: Account.put_code(state_with_account, address, code), else: state_with_account
+      case code do
```

Comment:
what if you wrote the commit_account_cache something like:
```
defp commit_account_cache({address, {:dirty, account = nil, code}}, state), do: Account.del_account(state, address)
defp commit_account_cache({address, {:dirty, account, code = {:dirty, code}}}, state) do 
  state_with_account = Account.put_account(state, address, account)
  Account.put_code(state_with_account, address, code)
end
defp commit_account_cache({address, {:dirty, account, code}}, state) do 
  Account.put_account(state, address, account)
end
```

Reply:
@InoMurko done